src/components/edit-modal.js

```diff
@@ -30,29 +32,40 @@ class EditModal extends Component {
   handleSubmit = e => {
     e.preventDefault()
     let { fileName, isPublic } = this.state
-    const { selectedData, editFile } = this.props
+    const { editFile, activeItems } = this.props
 
     if (fileName) {
-      var formData = new FormData()
-      formData.append('file_name', fileName)
-      formData.append('is_public', isPublic)
-      editFile(selectedData.pk, formData, this.successCallback)
+      var formdata = new FormData()
+      formdata.append('file_name', fileName)
+      formdata.append('is_public', isPublic)
+      editFile(activeItems[0].obj.id, formdata, this.successCallback)
     }
   }
-  handleCheck = () => {
+  handleCheckPublic = () => {
     this.setState({
       isPublic: !this.state.isPublic
     })
   }
+  handleCheckStar = () => {
+    this.setState({
+      starred: !this.state.starred
+    })
+  }
   successCallback = () => {
+    const id = this.props.currentFolder.id
+    this.props.getFolder(id)
     this.props.close()
+    this.setState({
+      fileName: '',
+      isPublic: false
+    })
   }
-  render() {
-    const { fileName, isPublic } = this.state
-    const { isLoading, showModal, close } = this.props
+  render () {
+    const { fileName, isPublic, starred } = this.state
+    const { updateFilePending, showModal, close } = this.props
     return (
       <Modal
-        size="large"
+        size='large'
         open={showModal}
```

Comment:
edit-file is almost as upload-file so try to merge these into one component

Reply:
Yea seems like a nice idea ... but i was thinking if we could use common edit-modal for file and folder, and common upload component for folder and file .